nncf/experimental/torch2/function_hook/serialization.py

```diff
@@ -101,11 +104,19 @@ def load_from_config(model: TModel, config: Dict[str, Any]) -> TModel:
     :return: The compressed model.
     """
     wrapped_model = wrap_model(model)
+
+    device = None
+    if not is_multidevice(wrapped_model):
+        device = get_model_device(wrapped_model)
+    else:
+        nncf_logger.warning("Model is on multiple devices. Cannot determine device for loaded modules.")
+
     hook_storage = get_hook_storage(wrapped_model)
     transformation_commands = cast(List[S_COMMAND], config[COMPRESSION_STATE_ATTR])
     for command in transformation_commands:
         module_cls = COMPRESSION_MODULES.get(command["module_cls_name"])
         module = module_cls.from_config(command["module_config"])
+        module.to(device)
```

Comment:
I encountered the same problem with the model on various devices:
https://github.com/openvinotoolkit/nncf/pull/3341/files#diff-f257219eb5631518ca7ec37bb9858c77491049dc999ca14d845a46517f263900R135-R136

Is there a reliable method to match the compression module from the insertion command to the target layer and its parameters? With this matching, it would be possible to determine the device.

Reply:
Currently we have limited support for multidevice models.
For not multi device model it works.

Reply:
I did not look at it. Currently multidevice support for new tracing works similar like for current.